### PR TITLE
doc: convert config guide to reST, document name restrictions for `builds:`

### PR DESCRIPTION
**Pull request by victorlin**
## Description of proposed changes

This PR adds a warning message to the **builds** section:

> image

This is documented internally with [workflow config schema][1] and [various wildcard constraints][2].

[1]: https://github.com/nextstrain/ncov/blob/464041debc66a86a33d3c1c697026a1adeb1d326/workflow/schemas/config.schema.yaml
[2]: https://github.com/nextstrain/ncov/blob/464041debc66a86a33d3c1c697026a1adeb1d326/Snakefile#L132-L138

To add the warning block, I first had to convert to reStructuredText.

## Related issue(s)

- #789
- #482

## Testing

_N/A_

## Release checklist

_N/A_

### CODE REVIEW

**Comment by jameshadfield** [comment]
Thanks @victorlin! Looks good. Perhaps explicitly call out the fact that they cannot have a full stop?

Also, the schema should be updated to avoid the build name ending with `_measurements` or `_root-sequence`, both of which are reserved for sidecar files. But this can be another PR / issue.

John & I talked about re-organising this file, and I'll try to do this in a separate PR on top of the tutorial branch.

File: docs/src/reference/configuration.rst [hunk 1]
```diff
@@ -0,0 +1,1033 @@
...
+-  type: object
+-  description: Named builds to produce by the workflow from the given inputs. Builds are indexed by name and include any number of build attributes that can be used to control subsampling, Auspice configuration, and more.
+
+.. warning::
+
+   Build names currently only allow alpha characters, underscores, and hyphens (``A-Z``, ``a-z``, ``_``, ``-``), but must not contain ``tip-frequencies`` as it is a special string used for Nextstrain builds.
+
+   Note that these are not allowed:
+
+   - Periods (``.``)
```
Comment (victorlin) [inline docs/src/reference/configuration.rst:71]: _from https://github.com/nextstrain/ncov/pull/896#issuecomment-1080134719_

> @jameshadfield: Perhaps explicitly call out the fact that they cannot have a full stop?

Added here along with some other explicit examples.

File: docs/src/reference/configuration.rst [hunk 2]
```diff
@@ -0,0 +1,1033 @@
...
+
+builds
+------
+
+-  type: object
+-  description: Named builds to produce by the workflow from the given inputs. Builds are indexed by name and include any number of build attributes that can be used to control subsampling, Auspice configuration, and more.
+
+.. warning::
+
+   Build names currently only allow alpha characters, underscores, and hyphens (``A-Z``, ``a-z``, ``_``, ``-``), but must not contain ``tip-frequencies`` as it is a special string used for Nextstrain builds.
```
Comment (victorlin) [inline docs/src/reference/configuration.rst:67]: I also noticed that just `tip-frequencies` passes the regex, is this fine? Or should it be matching for `_tip-frequencies`?

Comment (jameshadfield) [reply docs/src/reference/configuration.rst:67]: We should err on the side of caution here and make sure the build name doesn't end with [any of the 5 suffixes we filter out in the nextstrain server](https://github.com/nextstrain/nextstrain.org/blob/d3a1fb7185ca3348c4121535484145db4619b714/src/utils/index.js#L83-L88): 

```js

  // All JSON files which aren't a sidecar file with a known suffix are assumed to
  // be v2+ JSONs (aka "unified" JSONs)
  const sidecarSuffixes = ["meta", "tree", "root-sequence", "seq", "tip-frequencies", "measurements"];
  const datasets = jsonFiles
    .filter((filename) => !sidecarSuffixes.some((suffix) => filename.endsWith(`_${suffix}.json`)))
```

Comment (victorlin) [reply docs/src/reference/configuration.rst:67]: Added this to issue #900

**Pull request merged**